odxtools/diagcodedtype.py

```diff
@@ -142,6 +142,8 @@ def _encode_internal_value(
         """Convert the internal_value to bytes."""
         # Check that bytes and strings actually fit into the bit length
         if base_data_type == DataType.A_BYTEFIELD:
+            if isinstance(internal_value, bytearray):
```

Comment:
@andlaus are we accepting bytearray again?

Reply:
it would be nice, because mypy allows `bytearray` where `bytes` is specified. the problem is that `bitstruct.c` currently only accepts `bytearray` for the `unpack()` function, not for `pack()`. I suppose I should fix that...